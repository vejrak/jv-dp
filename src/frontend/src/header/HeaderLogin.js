// @flow

import React from 'react'
import Link from 'next/link'
import { logout } from '../user/actions'

type Props = $ReadOnly<{|
  isLoggedIn: boolean,
  logout: typeof logout,
  username: string,
|}>

class HeaderLogin extends React.PureComponent<Props> {
  handleLogout = () => {
    this.props.logout()
  }

  render() {
    const { isLoggedIn, username } = this.props

    return (
      <>
        {isLoggedIn && (
          <>
            <li className="nav-item">
              <span className="nav-link">Logged as {username}</span>
            </li>
            <li className="nav-item">
              <Link href="/changePassword">
                <a className="nav-link" href="#">
                  Change password
                </a>
              </Link>
            </li>
          </>
        )}
        <li className="nav-item">
          {isLoggedIn ? (
            <a className="nav-link" href="#" onClick={this.handleLogout}>
              Logout
            </a>
          ) : (
            <Link href="/login">
              <a className="nav-link">Login</a>
            </Link>
          )}
        </li>
      </>
    )
  }
}

export default HeaderLogin
