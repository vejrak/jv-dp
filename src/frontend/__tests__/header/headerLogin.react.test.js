import React from 'react'
import renderer from 'react-test-renderer'
import HeaderLogin from '../../src/header/HeaderLogin'

it('renders correctly with log in', () => {
  const tree = renderer
    .create(<HeaderLogin isLoggedIn logout={() => {}} name="test" />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly without log in', () => {
  const tree = renderer
    .create(<HeaderLogin isLoggedIn={false} logout={() => {}} name={null} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
