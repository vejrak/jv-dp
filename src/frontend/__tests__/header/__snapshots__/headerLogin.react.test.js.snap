// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with log in 1`] = `
Array [
  <li
    className="nav-item"
  >
    <span
      className="nav-link"
    >
      Logged as 
    </span>
  </li>,
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="#"
      onClick={[Function]}
    >
      Change password
    </a>
  </li>,
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="#"
      onClick={[Function]}
    >
      Logout
    </a>
  </li>,
]
`;

exports[`renders correctly without log in 1`] = `
<li
  className="nav-item"
>
  <a
    className="nav-link"
    href="/login"
    onClick={[Function]}
  >
    Login
  </a>
</li>
`;
